File: Benchmarking/src/gen267.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    std::vector<std::vector<s0>> p0;
    std::vector<std::vector<s0>> p1;
    s1(std::vector<std::vector<s0>> v0, std::vector<std::vector<s0>> v1): p0(v0), p1(v1) { }
  };
  struct s2 {
    std::vector<s1> p0;
    s0 p1;
    s2(std::vector<s1> v0, s0 v1): p0(v0), p1(v1) { }
  };
  double f30(const double &v0) {
    double v7 = v0;
    double v5 = v7;
    double v9 = v0;
    double v2 = v5;
    double v11 = v7;
    const std::vector<double> v15 { v5, v7, v2, v2, v5 };
    const double v21 = v15[3];
    std::vector<double> v18 = v15;
    const double v8 = v18[2];
    v7 = v11;
    v18[0] = v2;
    v18[4] = v21;
    v18[4] = v9;
    v18[3] = v9;
    return v8;
  }
  __attribute__((noinline))
  double f0(const s1 &v0, const s0 &v1, const s2 &v2, const double &v3) {
    const double v18 = v3 + v3;
    double v17 = v18;
    double v14 = v17;
    const double v19 = v3 / v17;
    v14 = v19;
    const double v47 = f30(v14);
    v17 = v3;
    return v47;
  }
  int main() {
    s1 v0({ { { { { { 0.0 }, { 1.0 } }, { { 2.0 }, { 3.0 } } } } }, { { { { { 4.0 }, { 5.0 } }, { { 6.0 }, { 7.0 } } } }, { { { { 8.0 }, { 9.0 } }, { { 10.0 }, { 11.0 } } } } } });
    s0 v1({ { { 12.0 }, { 13.0 } }, { { 14.0 }, { 15.0 } } });
    s2 v2({ { { { { { { { 16.0 }, { 17.0 } }, { { 18.0 }, { 19.0 } } } } }, { { { { { 20.0 }, { 21.0 } }, { { 22.0 }, { 23.0 } } } }, { { { { 24.0 }, { 25.0 } }, { { 26.0 }, { 27.0 } } } } } }, { { { { { { 28.0 }, { 29.0 } }, { { 30.0 }, { 31.0 } } } } }, { { { { { 32.0 }, { 33.0 } }, { { 34.0 }, { 35.0 } } } }, { { { { 36.0 }, { 37.0 } }, { { 38.0 }, { 39.0 } } } } } } }, { { { 40.0 }, { 41.0 } }, { { 42.0 }, { 43.0 } } } });
    double v3(44.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2, v3);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen267.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: [[s0]]
    var p1: [[s0]]
  }
  struct s2 {
    var p0: [s1]
    var p1: s0
  }
  func f30(_ v0: Double) -> Double {
    var v7: Double = v0
    var v5: Double = v7
    var v9: Double = v0
    var v2: Double = v5
    var v11: Double = v7
    let v15: [Double] = [v5, v7, v2, v2, v5]
    let v21: Double = v15[3]
    var v18: [Double] = v15
    let v8: Double = v18[2]
    v7 = v11
    v18[0] = v2
    v18[4] = v21
    v18[4] = v9
    v18[3] = v9
    return v8
  }
  @inline(never)
  func f0(_ v0: s1, _ v1: s0, _ v2: s2, _ v3: Double) -> Double {
    let v18: Double = v3 + v3
    var v17: Double = v18
    var v14: Double = v17
    let v19: Double = v3 / v17
    v14 = v19
    let v47: Double = f30(v14)
    v17 = v3
    return v47
  }
  func benchmark() {
    let v0: s1 = s1(p0: [[s0(p0: [[0.0], [1.0]], p1: [[2.0], [3.0]])]], p1: [[s0(p0: [[4.0], [5.0]], p1: [[6.0], [7.0]])], [s0(p0: [[8.0], [9.0]], p1: [[10.0], [11.0]])]])
    let v1: s0 = s0(p0: [[12.0], [13.0]], p1: [[14.0], [15.0]])
    let v2: s2 = s2(p0: [s1(p0: [[s0(p0: [[16.0], [17.0]], p1: [[18.0], [19.0]])]], p1: [[s0(p0: [[20.0], [21.0]], p1: [[22.0], [23.0]])], [s0(p0: [[24.0], [25.0]], p1: [[26.0], [27.0]])]]), s1(p0: [[s0(p0: [[28.0], [29.0]], p1: [[30.0], [31.0]])]], p1: [[s0(p0: [[32.0], [33.0]], p1: [[34.0], [35.0]])], [s0(p0: [[36.0], [37.0]], p1: [[38.0], [39.0]])]])], p1: s0(p0: [[40.0], [41.0]], p1: [[42.0], [43.0]]))
    let v3: Double = 44.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 44.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2, v3)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
